infrastructure/templates/public-api/main.bicep

```diff
@@ -0,0 +1,362 @@
+param createdBy string = 'Unknown'
+@description('Tag Value - Enter the name of the repo that the deployment script for the component name be found. If the component is deployed manually, the value should be N/A')
+param deploymentRepo string = 'N/A'
+
+
+//Networking Params --------------------------------------------------------------------------
+@description('Networking : Deploy subnets for networking')
+param deploySubnets bool = true
+
+@description('Networking : Included whitelist')
+param storageFirewallRules array = ['0.0.0.0/0']
+
+//Storage Params ------------------------------------------------------------------------------
+@description('Storage : Name of the root fileshare folder name')
+@minLength(3)
+@maxLength(63)
+param fileShareName string = 'data'
+
+@description('Storage : Type of the file share quota')
+param fileShareQuota int = 1
+
+//PostgreSQL Database Params -------------------------------------------------------------------
+@description('Database : administrator login name')
+@minLength(1)
+param dbAdministratorLoginName string = 'PostgreSQLAdmin'
+
+@description('Database : administrator password')
+@minLength(8)
+@secure()
+param dbAdministratorLoginPassword string //P$&RW6*h2V1CKwTm
+
+@description('Database : Azure Database for PostgreSQL sku name ')
+@allowed([
+  'Standard_B1ms'
+  'Standard_D4ads_v5'
+  'Standard_E4ads_v5'
+])
+param skuName string = 'Standard_B1ms'
+
+@description('Database : Azure Database for PostgreSQL Storage Size in GB')
+param storageSizeGB int = 32
+
+@description('Database : Azure Database for PostgreSQL Autogrow setting')
+param autoGrowStatus string = 'Disabled'
+
+//Container Registry Params ----------------------------------------------------------------
+@minLength(5)
+@maxLength(50)
+@description('Registry : Name of the azure container registry (must be globally unique)')
+param containerRegistryName string = 'eesapiacr'
+
+//Container App Params
+@description('Container App : Specifies the container image to deploy from the registry <name>:<tag>.')
+param acrHostedImageName string = 'hello-world'
+
+@description('Specifies the container port.')
+param targetPort int = 80
+
+@description('Container App : Specifies the container image to seed to the ACR.')
+param containerSeedImage string = 'mcr.microsoft.com/azuredocs/aci-helloworld'
+
+@description('Select if you want to seed the ACR with a base image.')
+param seedRegistry bool = true
+
+//ServiceBus Queue Params -------------------------------------------------------------------
+@description('Name of the Service Bus namespace')
+param namespaceName string = 'etlnamespace'
+
+@description('Name of the Queue')
+param queueName string = 'etlfunctionqueue'
+
+//ETL Function Paramenters ------------------------------------------------------------------
+@description('Application : Insights name')
+param applicationInsightsName string = 'etlFunctionInsights'
```

Comment:
Don't think we need this. As per another comment, names for Application Insights should just be derived from the app that it's attached to